DOCSUP-12328: Edit and translate to Russian (update materialize_mysql) (#27139)
ClickHouse/ClickHouse: docs/en/sql-reference/functions/array-functions.md

## empty {#function-empty}

Checks whether the input array is empty.

**Syntax**

``` sql
empty([x])
```

An array is considered empty if it contains all empty elements.

!!! note "Note"

>     Can be optimized by enabling the [optimize_functions_to_subcolumns](../../operations/settings/settings.md#optimize-functions-to-subcolumns) setting. With `optimize_functions_to_subcolumns = 1` the function reads only [size0](../../sql-reference/data-types/array.md#array-size) subcolumn instead of reading and processing the whole array column. The query `SELECT empty(arr) FROM table;` transforms to `SELECT arr.size0 = 0 FROM TABLE;`.

Reviewer: ```suggestion
    Can be optimized by enabling the [optimize_functions_to_subcolumns](../../operations/settings/settings.md#optimize-functions-to-subcolumns) setting. With `optimize_functions_to_subcolumns = 1` the function reads only [size0](../../sql-reference/data-types/array.md#array-size) subcolumn instead of reading and processing the whole array. The query `SELECT empty(arr) FROM table;` transforms to `SELECT arr.size0 = 0 FROM table;`.
```

The function also works for [strings](string-functions.md#empty) or [UUID](uuid-functions.md#empty).

**Arguments**

-   `[x]` — Input array. [Array](../data-types/array.md).

**Returned value**

-   Returns `1` for an empty array or `0` for a non-empty array.

Type: [UInt8](../data-types/int-uint.md).

**Example**

Query:

```sql
SELECT empty([]);
```

Result:

```text
┌─empty(array())─┐
│              1 │
└────────────────┘
```

## notEmpty {#function-notempty}

Checks whether the input array is non-empty.

**Syntax**

``` sql
notEmpty([x])
```
An array is considered non-empty if it contains at least one non-empty element.

!!! note "Note"

>     Can be optimized by enabling the [optimize_functions_to_subcolumns](../../operations/settings/settings.md#optimize-functions-to-subcolumns) setting. With `optimize_functions_to_subcolumns = 1` the function reads only [size0](../../sql-reference/data-types/array.md#array-size) subcolumn instead of reading and processing the whole array column. The query `SELECT notEmpty(arr) FROM table` transforms to `SELECT arr.size0 != 0 FROM TABLE`.

Reviewer: ```suggestion
    Can be optimized by enabling the [optimize_functions_to_subcolumns](../../operations/settings/settings.md#optimize-functions-to-subcolumns) setting. With `optimize_functions_to_subcolumns = 1` the function reads only [size0](../../sql-reference/data-types/array.md#array-size) subcolumn instead of reading and processing the whole array. The query `SELECT notEmpty(arr) FROM table` transforms to `SELECT arr.size0 != 0 FROM table`.
```

## empty {#function-empty}

Checks whether the input array is empty.

**Syntax**

``` sql
empty([x])
```

> An array is considered empty if it contains all empty elements.

Reviewer 2: ```suggestion
An array is considered empty if it does not contain any elements.```

!!! note "Note"
    Can be optimized by enabling the [optimize_functions_to_subcolumns](../../operations/settings/settings.md#optimize-functions-to-subcolumns) setting. With `optimize_functions_to_subcolumns = 1` the function reads only [size0](../../sql-reference/data-types/array.md#array-size) subcolumn instead of reading and processing the whole array column. The query `SELECT empty(arr) FROM TABLE;` transforms to `SELECT arr.size0 = 0 FROM TABLE;`.

The function also works for [strings](string-functions.md#empty) or [UUID](uuid-functions.md#empty).

**Arguments**

-   `[x]` — Input array. [Array](../data-types/array.md).

**Returned value**

-   Returns `1` for an empty array or `0` for a non-empty array.

Type: [UInt8](../data-types/int-uint.md).

**Example**

Query:

```sql
SELECT empty([]);
```

Result:

```text
┌─empty(array())─┐
│              1 │
└────────────────┘
```

## notEmpty {#function-notempty}

Checks whether the input array is non-empty.

**Syntax**

``` sql
notEmpty([x])
```

> An array is considered non-empty if it contains at least one non-empty element.

Reviewer 2: ```suggestion
An array is considered non-empty if it contains at least one element.
```
